tools/nl/embeddings/utils.py

```diff
@@ -214,26 +211,35 @@ def _download_model_from_gcs(ctx: Context, model_folder_name: str) -> str:
   return os.path.join(local_dir, model_folder_name)
 
 
-def build_embeddings(ctx, texts: List[str], dcids: List[str]) -> pd.DataFrame:
+def build_embeddings(ctx, text2sv: Dict[str, str]) -> pd.DataFrame:
   """Builds the embeddings dataframe.
-  
-  Texts and dcids should be of equal length such that
-  a text at a given index corresponds to the dcid at that index.
 
   The output dataframe contains the embeddings columns (typically 384) + dcid + sentence.
   """
-  assert len(texts) == len(dcids)
+  texts = sorted(list(text2sv.keys()))
 
-  embeddings = ctx.model.encode(texts, show_progress_bar=True)
+  if ctx.model:
+    embeddings = ctx.model.encode(texts, show_progress_bar=True)
+  else:
+    embeddings = []
+    for i, chuck in enumerate(chunk_list(texts, _CHUNK_SIZE)):
+      logging.info('texts %d to %d', i * _CHUNK_SIZE, (i + 1) * _CHUNK_SIZE)
```

Comment:
I think either: 
`i * _CHUNK_SIZE` should be `i * _CHUNK_SIZE + 1`
or `(i + 1) * _CHUNK_SIZE` should be `(i + 1) * _CHUNK_SIZE - 1`